Codegen/src/Nodes/TypedDataNode.cs

```diff
@@ -924,19 +926,97 @@ private static System.Text.StringBuilder GetIndent(int level, int indent, bool i
         public static List<ITextLine> InsertChildren(ColumnsLayout? layout, List<string> rootProcedures, List< Tuple<string,string> > rootVariableName, DataDefinition ownerDefinition, DataDefinition type, int level, int indent)
         {
             var lines = new List<ITextLine>();
+            // List of all the CopyDirectives that have been added to the lines
+            List<CopyDirective> usedCopies = new List<CopyDirective>();
             foreach (var child in type.Children)
             {
-                bool bCanGenerate = !child.IsInsideCopy();
+                bool bIsInsideCopy = child.IsInsideCopy();
+
                 //Handle Typedef whose body is inside a COPY
                 if (child.IsFlagSet(Flag.InsideTypedefFromCopy))
                 {
                     if (child.IsFlagSet(Flag.IsTypedefCopyNode))
                     {
-                        lines.AddRange(child.Lines);
+                        if (child.Lines.Sum(l => l.Text.Count(c => c == '.')) > 1)
```

Comment:
Line added betwen 940 and 950 are strange, the way of counting the number of instructions by counting dot characters it is very strange, because a dot character can appear in a literal string.
I run the test called "TestCopyInsideTypeDefInsideDependency"  I never reached this point. 
I don't understand why it is necessary to translated multilines instructions into a singleline. In my opinion this code is not necessary and the previous code can be kept.

Reply:
See last commits